mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala

```diff
@@ -172,37 +274,77 @@ object StringIndexer extends DefaultParamsReadable[StringIndexer] {
 
   @Since("1.6.0")
   override def load(path: String): StringIndexer = super.load(path)
+
+  // Returns a function used to sort strings by frequency (ascending or descending).
+  // In case of equal frequency, it sorts strings by alphabet (ascending).
+  private[feature] def getSortFunc(
+      ascending: Boolean): ((String, Long), (String, Long)) => Boolean = {
+    if (ascending) {
+     (a: (String, Long), b: (String, Long)) => {
```

Comment:
I think indent if off one space here. Can you write something like `((strA, freqA), (strB, freqB)) =>` here for clarity? You could also write this comparison once and then wrap it in something that flips the comparison if `ascending` is `false`, but no big deal.

Reply:
I can only write it as a pattern matching anonymous function: `{ case (param1, param2) => ... }`, because Scala compiler complains `Tuples cannot be directly destructured in method or function parameters.`.